File: contest/scaler/algobeeII/H.py
def count_subarrays(arr):
    prefix_sum = 0
    count = 0
    # Initialize with 0 to handle cases where prefix sum is equal to 0
    prefix_counts = {0: 1}

    for num in arr:
        prefix_sum += num
        if prefix_sum >= 0:
            count += 1
        if prefix_sum in prefix_counts:
            count += prefix_counts[prefix_sum]
            prefix_counts[prefix_sum] += 1
        else:
            prefix_counts[prefix_sum] = 1

    return count


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        print(count_subarrays(arr))


main()

File: contest/scaler/algobeeII/H.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> arr(n);
        for (int& a : arr) {
            cin >> a;
        }

        // Prefix Sum
        vector<long long> prefixSum(n + 1, 0);
        for (int i = 0; i < n; i++) {
            prefixSum[i + 1] = prefixSum[i] + arr[i];
        }

        // Find the number of subarrays with sum greater than or equal to 0
        long long count = 0;
        map<long long, long long> prefixCounts;
        prefixCounts[0] = 1;

        for (int i = 1; i <= n; i++) {
            for (auto it = prefixCounts.lower_bound(prefixSum[i]); it != prefixCounts.end(); it++) {
                count += it -> second;
            }
            prefixCounts[prefixSum[i]]++;
        }

        cout << count << endl;
    }

    return 0;
}